httpx_ws/transport.py

```diff
@@ -218,10 +235,11 @@ async def _handle_ws_request(
         assert isinstance(request.stream, AsyncByteStream)
 
         self.scope = scope
-        self.exit_stack = contextlib.AsyncExitStack()
-        stream, accept_response = await self.exit_stack.enter_async_context(
-            ASGIWebSocketAsyncNetworkStream(self.app, self.scope)  # type: ignore[arg-type]
-        )
+        async with contextlib.AsyncExitStack() as stack:
+            stream, accept_response = await stack.enter_async_context(
+                ASGIWebSocketAsyncNetworkStream(self.app, self.scope, self._task_group)  # type: ignore[arg-type]
+            )
+            self.exit_stacks.append(stack.pop_all())
 
```

Comment:
From what I understand, the main thing is to create a task group at the top level. Could we simplify everything by opening a task group here, append it to the exit stacks and pass it to `ASGIWebSocketAsyncNetworkStream`? This would avoid to have the separate single `_exit_stack` and `__aenter__`/`__aexit__`.

Reply:
I tried what you suggested, but the new test failed. This PR is a bit old and I don't remember all the details, but feel free to push your changes if you think it can be simplified.